lib/appraisal/matrix/extensions/appraisal_file.rb

```diff
@@ -5,7 +5,26 @@
 
 module Appraisal::Matrix
   module AppraiseFileWithMatrix
-    include RubygemsHelper
+
+    class VersionArray
+      include RubygemsHelper
+
+      SUPPORTED_VERSION_STEPS = [:major, :minor, :patch].freeze
+
+      attr_reader :minimum_version, :maximum_version, :step
+
+      def initialize(min:, max: nil, step: :minor)
+        SUPPORTED_VERSION_STEPS.include?(step) or raise("Unsupported version step: #{step}")
+
+        @minimum_version = Gem::Version.new(min)
+        @maximum_version = max ? Gem::Version.new(max) : nil
+        @step = step.to_sym
+      end
+
+      def for_gem(gem_name)
```

Comment:
It feels to me that `gem_name` should be in the initialize method. We shouldn't be re-using this class with a different gem since each gem will have their own options.

This method would then make sense to just be named `versions`